src/main/java/org/assertj/core/api/AbstractInstantAssert.java

```diff
@@ -32,7 +32,7 @@
  * @since 3.7.0
  */
 public class AbstractInstantAssert<SELF extends AbstractInstantAssert<SELF>>
-    extends AbstractTemporalAssert<SELF, Instant> {
+    extends AbstractTemporalAssert<SELF, Instant> implements ComparableAssert<SELF, Instant> {
```

Comment:
try extending `AbstractComparableAssert<SELF, Instant>` to get the comparable assertions implementation for free.

Reply:
I tried doing that at the `AbstractTemporalAssert` level, but since `Temporal` doesn't extend `Comparable` that didn't work.  Seemed better to inherit the Temporal functionality and duplicate the Comparable, but I guess it could be the other way around?  Or is there a better class hierarchy modification you can think of?

Reply:
If a type does not implement `Comparable`, its assertion type should not implement `ComparableAssert`.

Moreover, I believe it is better to use `isAfter` instead of `isGreaterThan` for `Instant` assertions as `after` expresses time better than `greater` but I'm ok to add the comparable assertions to avoid breaking people code.

Reply:
I agree. Thanks.